### ISSUE TITLE
index undefined value

### ISSUE THREAD

**Issue by lolBig** [facebook/react-devtools#845]
```
let serverList = []
for (let i in worldServers) {
    serverList.push(
        <option value={i}>{worldServers[i].name}{worldServers[i].delay}</option>
    )
}
<select>{serverList}</select>
```

if worldServers[i].delay is undefined, will throw an exception in backend.js:8688.

**Comment by bvaughn** [comment]
More context here: the specific error is on this line:
```js
 if (!props && internalInstance._currentElement && internalInstance._currentElement.props) {
```
and is:
```
Uncaught TypeError: Cannot read property '_currentElement' of undefined
    at getData (backend.js:8688)
    at walkNode (backend.js:8576)
    at backend.js:8579
    at Array.forEach (<anonymous>)
    at walkNode (backend.js:8578)
    at backend.js:8579
    at Array.forEach (<anonymous>)
    at walkNode (backend.js:8578)
    at backend.js:8579
    at Array.forEach (<anonymous>)
```

**Comment by bvaughn** [comment]
The minimal case to reproduce this bug is the following:
```js
var node = document.getElementById('container');
ReactDOM.render(<option>foo{null}</option>, node);
```

**Comment by bvaughn** [comment]
Ping @flarnie in case you're interested in looking into this 😄  Else I will pick it up in an hour or so

**Comment by flarnie** [comment]
So far in Chrome I'm not able to reproduce this in a CRA app. Will try in Firefox next.
![screen shot 2017-07-13 at 10 43 45 am](https://user-images.githubusercontent.com/1114467/28179904-475a5d20-67b8-11e7-95cf-ad62047f69d2.png)

**Comment by flarnie** [comment]
Nevermind - it does reproduce with the exact examples mentioned.

**Comment by flarnie** [comment]
This is also stack-reconciler only - doesn't repro when running React v16.0.0-alpha.12. I'll keep fixing it so long as it seems easily fixable, but to me this is lower priority knowing that it is related to non-fiber code.

**Comment by flarnie** [comment]
I've also confirmed that just tweaking the conditionals in 'getData' to handle the 'null' value does not fix the issue. Looks like we don't expect to get 'null' as a possible value there, so something earlier on must be going wrong. Continuing to look into it.

**Comment by tifosiblack** [comment]
For me, the root cause it not that internalInstance is undefined, it's why we have an undefined child, in data.children retrieved by getData012 (or getData).

In fact I can simply reproduce the error when I have prototype extension of Array. 

```js
function childrenList(children) {
  var res = [];
  for (var name in children) {
    res.push(children[name]);
  }
  return res;
}
```
Using a for .. in without using hasOwnProperty is dangerous.

**Comment by bvaughn** [comment]
@tifosiblack You might be interested in checking out Flarnie's work to fix this on #853
